app/scripts/controllers/app-state.js

```diff
@@ -38,14 +38,14 @@ export default class AppStateController extends EventEmitter {
       recoveryPhraseReminderHasBeenShown: false,
       recoveryPhraseReminderLastShown: new Date().getTime(),
       outdatedBrowserWarningLastShown: new Date().getTime(),
-      collectiblesDetectionNoticeDismissed: false,
+      nftsDetectionNoticeDismissed: false,
```

Comment:
With these state changes, we'll want to create a migration to remove the old key.  It would look something like this:

https://github.com/MetaMask/metamask-extension/pull/17722/files#diff-174015d284d8348ea67e5895b9b7acd90ac08c2c0d0cdba0d352de2908bcb3b8